### ISSUE TITLE
New value for requirePaddingNewLinesInObjects

### ISSUE THREAD

**Issue by jmeas** [jscs-dev/node-jscs#1755]
(Note: this is not quite the same as #1551. That use case should be considered, too!)

Consider these two possibilities:

```js
View.extend({
  initialize: function() {},

  onDestroy: function() {}
});
```

```js
View.extend({

  initialize: function() {},

  onDestroy: function() {}

});
```

I've found that people either stick to one or the other (or don't care and mix them).

Afaict it's not possible to allow for **up to** a single newline padding while disallowing more than that.

One possibility, if #1551 is ignored, would be `"requirePaddingNewLinesInObjects": 1`, which would allow **up to** one newline.

*shrug* I'd use it!

(p.s. forgive me if an issue already exists/was closed in regards to this. I ran a search and nothing came up).

**Comment by markelog** [comment]
I guess new rule would be a better choice, would you like to implement it?

**Comment by jmeas** [comment]
I don't foresee myself having time in the near future, but if I do then I would definitely like to figure out how to implement it.

**Comment by Nargonath** [comment]
@jmeas Are you working on it at the moment?

**Comment by jmeas** [comment]
I am not

**Comment by Nargonath** [comment]
@jmeas do you mind if I take a shot at it? Or do you prefer to save it for later?

**Comment by jmeas** [comment]
Go for it!

**Comment by Nargonath** [comment]
@jmeas Alright, thanks.

**Comment by Nargonath** [comment]
Just to be sure I understand what you want in the rule, you want the rule to allow up to 2 line breaks?

These cases would be possible:
* `disallowMoreNewlinesPaddingInObjects: 1`
````javascript
View.myFunction({
    prop1: function() {},

    prop2: function() {}
});
````

* `disallowMoreNewlinesPaddingInObjects: 2`
````javascript
View.myFunction({

    prop1: function() {},

    prop2: function() {}
});
````

**Comment by hzoo** [comment]
We released [3.0](https://github.com/jscs-dev/node-jscs/releases/tag/v3.0.0) recently! We also wrote a [post](https://medium.com/@markelog/jscs-end-of-the-line-bc9bf0b3fdb2#.oot3wvgpu) about how 3.0 will be our last major release (since we are deprecating the project) and that we are [merging with ESLint](http://eslint.org/blog/2016/04/welcoming-jscs-to-eslint)!

Thus we won't be working on any new features moving forward which includes new rules/options. We will continue to fix bugs and work on CST/better autofixing for current rules in the upcoming months both to help JSCS users and to gain experience that we can hopefully use in ESLint. But otherwise we both teams will be working on getting [ESLint feature parity with JSCS](https://github.com/eslint/eslint/milestones/JSCS%20Compatibility) and just a nice transition for our users.

I'm putting an `orphaned` label to note this issue isn't planned to be worked on, but might be a good contribution as a custom rule (not in JSCS core itself).

I would encourage anyone who wants the rule to try writing it themselves: I would either write a custom rule or submit an issue to ESLint and if accepted a PR. Otherwise a custom rule that won't be in core repo itself.

JSCS rules are also very similar to ESLint rules: here are some resources to get started: 

- [Project-specific Lint Rules with ESLint](https://gist.github.com/jareware/7179093)
- [Creating Custom Rules for ESLint](https://insideops.wordpress.com/2015/12/08/creating-custom-rules-for-eslint)
- [Maintaining JavaScript Code Quality with ESLint ](https://www.paypal-engineering.com/2014/12/12/maintaining-javascript-code-quality-with-eslint/)
